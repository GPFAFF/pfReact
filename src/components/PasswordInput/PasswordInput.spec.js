import React from 'react';
import renderer from 'react-test-renderer';
import PasswordInput from './PasswordInput';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

test('toggles input type when show/hide password is clicked', () => {
  const wrapper = shallow(
    <PasswordInput
      htmlId="test"
      name="test"
      value=""
      onChange={() => {}}
      showVisibilityToggle
    />
  );

  expect(wrapper.find({type: 'password'})).toHaveLength(1);
  expect(wrapper.find({type: 'text'})).toHaveLength(0);

  wrapper.find('button').simulate('click');

  expect(wrapper.find({type: 'password'})).toHaveLength(0);
  expect(wrapper.find({type: 'text'})).toHaveLength(1);
});

test('hides password quality by default', () => {
  const tree = renderer.create(
    <PasswordInput
      htmlId='test'
      name='test'
      onChange={() => {}}
      value='Uisi37#8iad'
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('shows password quality when enabled and password is entered', () => {
  const tree = renderer.create(
    <PasswordInput
      htmlId='test'
      name='test'
      onChange={() => {}}
      showQuality
      value='Uisi37#8iad'
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
