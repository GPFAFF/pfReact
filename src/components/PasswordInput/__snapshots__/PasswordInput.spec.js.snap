// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hides password quality by default 1`] = `
<div
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <label
    className="Label"
    htmlFor="test"
  >
    <span
      className="Label__text"
    >
      Password
    </span>
    <span
      className="Label__required"
    >
      *
    </span>
  </label>
  <input
    id="test"
    maxLength={50}
    name="test"
    onChange={[Function]}
    type="password"
    value="Uisi37#8iad"
  />
</div>
`;

exports[`shows password quality when enabled and password is entered 1`] = `
<div
  style={
    Object {
      "marginBottom": 16,
    }
  }
>
  <label
    className="Label"
    htmlFor="test"
  >
    <span
      className="Label__text"
    >
      Password
    </span>
    <span
      className="Label__required"
    >
      *
    </span>
  </label>
  <input
    id="test"
    maxLength={50}
    name="test"
    onChange={[Function]}
    showQuality={true}
    type="password"
    value="Uisi37#8iad"
  />
</div>
`;
